addon/decorator.ts

```diff
@@ -3,15 +3,16 @@ import isElementDescriptor from './-private/is-element-descriptor';
 import throwConsoleError from './-private/throw-console-error';
 import Component from '@glimmer/component';
 import PropTypes, { Validator } from 'prop-types';
-import config from 'ember-get-config';
+import { getOwner } from '@ember/application';
 import { isNone } from '@ember/utils';
 import { closest } from './-private/closest-string';
 import { macroCondition, isDevelopingApp } from '@embroider/macros';
 
 const REGISTERED_ARGS = Symbol('args');
 const INTERCEPT_CLASS = 'ForbidExtraArgsIntercept';
 
-function shouldThrowErrors(): boolean {
+function shouldThrowErrors(owner: any): boolean {
```

Comment:
I can also pass along the actual config instead of the owner, maybe that's less "weird", though that's a bit more work on the invocation side.